src/providers/postgres/qgspostgresdataitemguiprovider.cpp

```diff
@@ -1008,3 +1036,87 @@ void QgsPostgresDataItemGuiProvider::moveProjectToSchema( QgsPGProjectItem *proj
     conn->unref();
   }
 }
+
+void QgsPostgresDataItemGuiProvider::moveLayerToSchema( QgsPGLayerItem *layerItem, QgsDataItemGuiContext context )
+{
+  QgsPGSchemaItem *schemaItem = qobject_cast<QgsPGSchemaItem *>( layerItem->parent() );
+  if ( !schemaItem )
+  {
+    return;
+  }
+
+  std::unique_ptr<QgsAbstractDatabaseProviderConnection> dbConnection( schemaItem->databaseConnection() );
+  if ( !dbConnection )
+  {
+    return;
+  }
+
+  QgsDatabaseSchemaSelectionDialog *dlg = new QgsDatabaseSchemaSelectionDialog( dbConnection.release() );
+
+  if ( dlg->exec() == QDialog::Accepted )
+  {
+    const QgsDataSourceUri dsUri( layerItem->uri() );
+    QgsPostgresConn *conn = QgsPostgresConn::connectDb( dsUri, false );
+    if ( !conn )
+    {
+      notify( tr( "Move Layer to Another Schema" ), tr( "Unable to move layer to another schema." ), context, Qgis::MessageLevel::Warning );
+      return;
+    }
+
+    const QgsPostgresLayerProperty &layerInfo = layerItem->layerInfo();
+    const QString newSchemaName = dlg->selectedSchema();
+    if ( newSchemaName == layerInfo.schemaName )
+    {
+      notify( tr( "Move Layer to Another Schema" ), tr( "Cannot copy to the schema where the layer already is." ), context, Qgis::MessageLevel::Warning );
+      conn->unref();
+      return;
+    }
+
+    QString typeOfAlter;
```

Comment:
This method should ideally be exposed by QgsAbstractDatabaseProviderConnection, so that it's accessible to plugins & other database providers too.

Reply:
Hopefully I managed to do it correctly. Feel free to point out what needs to be fixed or reworked.